pkg/sql/sqlstats/persistedsqlstats/compaction_exec.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package persistedsqlstats
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/scheduledjobs"
+	"github.com/cockroachdb/cockroach/pkg/security"
+	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlutil"
+	"github.com/cockroachdb/errors"
+)
+
+// StatsCompactor is responsible for compacting older SQL Stats. It is
+// executed by sql.sqlStatsCompactionResumer.
+type StatsCompactor struct {
+	Settings *cluster.Settings
+
+	initialized bool
+	db          *kv.DB
+	ie          sqlutil.InternalExecutor
+
+	TestingKnobs *StatsCompactorTestingKnobs
+}
+
+// StatsCompactorTestingKnobs can be used to tune the behavior of
+// StatsCompactor in unit tests.
+type StatsCompactorTestingKnobs struct {
+	// DisableFollowerRead disables the use of follower read used in
+	// StatsCompactor. This is used because follower can cause problems in
+	// in unit tests where we immediately start running follower read SQL queries
+	// on system tables that are yet to be created. This is not a scenario that is
+	// possible in real life.
+	DisableFollowerRead bool
+
+	// JobSchedulerEnv overrides the environment to use for scheduled jobs.
+	JobsSchedulerEnv scheduledjobs.JobSchedulerEnv
+}
+
+// Init initializes the StatsCompactor. This method needs to be called before
+// the DeleteOldestEntries method.
+func (c *StatsCompactor) Init(
```

Comment:
Why `Init` and not just `NewStatsCompactor` so you don't need to stress about whether or not it has been initialized?